client/src/hooks/use-router.tsx

```diff
@@ -69,6 +69,13 @@ export function useRouter() {
                 return { Component: DataSources };
               },
             },
+            {
+              path: 'add-data-source',
```

Comment:
if we do a modal we don't need a path

Reply:
Planning to do a modal in a different PR with the modal extraction to stop this PR from getting too messy